src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSource.cs

```diff
@@ -860,6 +890,48 @@ private void OnSuggestedActionsChanged(Workspace currentWorkspace, DocumentId cu
 
                 Volatile.Write(ref _lastSolutionVersionReported, solutionVersion);
             }
+
+            public async Task<ISuggestedActionCategorySet> GetSuggestedActionCategoriesAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+            {
+                var provider = _owner;
+                var result = new List<string>();
+                using (var asyncToken = _owner.OperationListener.BeginAsyncOperation(nameof(GetSuggestedActionCategoriesAsync)))
+                {
+                    var selection = await GetSpanAsync(range).ConfigureAwait(false);
+                    if (selection != null)
+                    {
+                        var document = range.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                        if (await HasFixesAsync(provider, document, range, cancellationToken).ConfigureAwait(false))
+                        {
+                            result.Add(PredefinedSuggestedActionCategoryNames.CodeFix);
+                            result.Add(PredefinedSuggestedActionCategoryNames.ErrorFix);
```

Comment:
What this means is that if there are only 'warning' severity fixes available, GetSuggestedActionCategoriesAsync() will return {CodeFix, ErrorFix}, and we'll put up the error lightbulb. When the user invokes Ctrl+. however, we call GetSuggestedActions() and receive a set of actions with category 'CodeFix', and we'll display the regular lightbulb instead. If the lightbulb was already visible, the icon will change.

GetSuggestedActionCategoriesAsync() should only return a category if it has applicable actions that will be returned for it on the call to GetSuggestedActions().